src/App.tsx

```diff
@@ -66,28 +57,21 @@ export const App: React.FC = () => {
                 {author && !loaded && (
                   <Loader />
                 )}
-
-                {author && loaded && hasError && (
+                {hasErrorCondition && (
                   <div
                     className="notification is-danger"
                     data-cy="PostsLoadingError"
                   >
                     Something went wrong!
                   </div>
                 )}
-
-                {author && loaded && !hasError && posts.length === 0 && (
+                {hasNoPostsCondition && (
```

Comment:
Appreciate your approach here: having `hasNoPostsCondition` extracted as a variable is much better than writing the whole `author && loaded && !hasError && !posts.length;` thing here.